sql/core/src/test/scala/org/apache/spark/sql/DataFrameRangeSuite.scala

```diff
@@ -113,11 +113,7 @@ class DataFrameRangeSuite extends QueryTest with SharedSQLContext with Eventuall
     val random = new Random(seed)
 
     def randomBound(): Long = {
-      val n = if (random.nextBoolean()) {
-        random.nextLong() % (Long.MaxValue / (100 * MAX_NUM_STEPS))
-      } else {
-        random.nextLong() / 2
-      }
+      val n = random.nextLong() % (Long.MaxValue / (100 * MAX_NUM_STEPS))
```

Comment:
why change this?

Reply:
Because there may have been overflows here

Reply:
which step can cause overflow before?

Reply:
Well, in case you just do `random.nextLong() / 2`, if the random long is close to the max long, its easy that the sum causes an overflow

Reply:
But this changes the distribution of the randomly generated long values, which may reduce the test coverage. Does it cause overflow in many tests, or just a few?

Reply:
It failed constantly..